src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/OnTypeRenameHandlerTest.cs

```diff
@@ -0,0 +1,278 @@
+        }
+
+        [Fact]
+        public async Task HandleRequestAsync_HtmlProjection_RemapsHighlightRange()
+        {
+            // Arrange
+            var called = false;
+            var expectedResponse = GetMatchingHTMLBracketRange(5);
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(Uri, Mock.Of<LSPDocumentSnapshot>(d => d.Version == 0));
+
+            var htmlResponse = GetMatchingHTMLBracketRange(10);
+            var requestInvoker = GetRequestInvoker<DocumentOnTypeRenameParams, DocumentOnTypeRenameResponseItem>(
+                htmlResponse,
+                (method, serverContentType, highlightParams, ct) =>
+                {
+                    Assert.Equal(MSLSPMethods.OnTypeRenameName, method);
+                    Assert.Equal(RazorLSPConstants.HtmlLSPContentTypeName, serverContentType);
+                    called = true;
+                });
+
+            var projectionResult = new ProjectionResult()
+            {
+                LanguageKind = RazorLanguageKind.Html,
+            };
+            var projectionProvider = GetProjectionProvider(projectionResult);
+
+            var documentMappingProvider = GetDocumentMappingProvider(expectedResponse.Ranges, 0, RazorLanguageKind.Html);
+
+            var onTypeRenameHandler = new OnTypeRenameHandler(documentManager, requestInvoker, projectionProvider, documentMappingProvider);
+            var onTypeRenameRequest = new DocumentOnTypeRenameParams()
+            {
+                TextDocument = new TextDocumentIdentifier() { Uri = Uri },
+                Position = new Position(10, 5)
+            };
+
+            // Act
+            var result = await onTypeRenameHandler.HandleRequestAsync(onTypeRenameRequest, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.True(called);
+            Assert.Collection(result.Ranges,
+                r => Assert.Equal(expectedResponse.Ranges[0], r),
+                r => Assert.Equal(expectedResponse.Ranges[1], r));
+        }
+
+        [Fact]
+        public async Task HandleRequestAsync_VersionMismatch_DiscardsLocation()
+        {
+            // Arrange
+            var called = false;
+            var expectedResponse = GetMatchingHTMLBracketRange(5);
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(Uri, Mock.Of<LSPDocumentSnapshot>(d => d.Version == 1));
+
+            var htmlResponse = GetMatchingHTMLBracketRange(10);
+            var requestInvoker = GetRequestInvoker<DocumentOnTypeRenameParams, DocumentOnTypeRenameResponseItem>(
+                htmlResponse,
+                (method, serverContentType, highlightParams, ct) =>
+                {
+                    Assert.Equal(MSLSPMethods.OnTypeRenameName, method);
+                    Assert.Equal(RazorLSPConstants.HtmlLSPContentTypeName, serverContentType);
+                    called = true;
```

Comment:
I'm probably guilty of this one too but a more descriptive name would be helpful, like `invokerCalled` or something.

Reply:
ditto for the other places with that variable.